GHzDACs/GHzAD_v7.md

```diff
@@ -0,0 +1,437 @@
+    FPGA + electronics in and out:      6.8 ns
+    Cable delay:                        1.4 ns/ft
+
+                Time delay      0 clock delay   1 clock delay       2 clock delay
+    1’ cable    8.2 ns          0-122 MHz       122-200 MHz
+    3’ cable    11.0 ns         0-91 MHz        91-182 MHz          182-200 MHz
+
+See GHzDAC for details of up daisy chain.
+For the AD measurement bits, the signals are output at Daisyout[0,1]  according to the following:
+
+The output bitstream in the daisy chain will be a simple 1+8 bit serial protocol
+Daisyout[0]     ...0 0 0 1 b0 b2 b4 b6 b8 b10 b12 b14 0 0 0 ...
+Daisyout[1]     ...0 0 0 1 b1 b3 b5 b7 b9 b11 b13 b15 0 0 0 ...
+The bits b0...b15 are 16 output bits of the AD converter, demondulator, and thresholding detector.
+Note the 0 to 1 transition is the start of the serial transmission.    
+
+
+
+## Output Monitors
+
+### dipswitch DSW[7]=1
+
+    mon[0] = regdone = received Ethernet input for a register write
+    mon[1] = ethersend  =  sent Ethernet output initiated by a register readback
+
+2 clock latency
+
+### dipswitch DSW[7]=0
+
+Outputs of mon0 and mon1 are controlled by registers `mon0[7..0]` and `mon1[7..0]` according to the following table. 3 clock latency
+
+    0   REGwrite        Ethernet in, write to registers
+    1   SRAMwrite       Ethernet in, write to SRAM memory
+    2   start           Start AD converter
+    3   aon             On when averaging 
+    4   don             On when demodulating
+    5   ADdone          ADconverter/demodulator done for one retrigger
+    6   alldone         All triggers done for one start
+    7   fifo_adv        Read new byte from fifo, 44 pulses total (named data_next)
+    8   fifostart       start pulse for fifo read (44 bytes) for packet send
+    9   spare 
+
+    10  DaisyIn[0]      DaisyIn[0] signal after fast input FF
+    11  DaisyIn[1]
+    12  DaisyIn[2]
+    13  DaisyIn[3]
+    14  DaisyOut[0]     DaisyOut[0] signal before fast output FF
+    15  DaisyOut[1]
+    16  DaisyOut[2]
+    17  DaisyOut[3]
+
+    18  d[0]            This is ADdemodulator output bits (bit readout mode)
+    19  d[1]
+    ...
+    25  d[7]
+    26  (d[8])          Not used now
+    ...
+    33  (d[15])
+
+
+## Initial Bringup
+    
+1. General functionality:
+  1. Power up, and check led(0) is off (FPGA locked), and led(1) is on (GHz PLL unlocked).
+  2. Write to board and read back, showing build number and Clock_Mux not locked.
+  3. Write to board to set PLL, and check led(1) goes off.  Use same PLL shift data as for GHzDAC.
+  4. Input AC signal into I and/or Q AD input, and check large (~0.5 V) signal causes led(2) for I and led(3) for Q to go on. This checks AD and over range function.
+2. Average mode:
+  5. Input as above, but send command for 1 average, auto start.  Look at downloaded data, and check that it oscillates properly.  Check that mon(1) goes high for 16 us after auto ethernet command is sent.  
```

Comment:
Probably only 8 us now.  Also, the monitors are switchable, so add a step to set the DIP switches or program the monitors as needed.